compiler/rustc_privacy/src/lib.rs

```diff
@@ -532,15 +551,15 @@ impl<'tcx> EmbargoVisitor<'tcx> {
         vis: ty::Visibility,
         module: LocalDefId,
     ) {
-        let level = Some(Level::Reachable);
-        if vis.is_public() {
-            self.update(def_id, level);
-        }
+        let private_vis = ty::Visibility::Restricted(self.tcx.parent_module_from_def_id(def_id));
+        let mut effective_vis = EffectiveVisibility::from_vis(private_vis);
+        effective_vis.set_public_at_level(Level::Reachable);
```

Comment:
I think the inherited effective vis here should be taken from the `macro` item instead of being constructed like this.